cache/rediscache/rediscache.go

```diff
@@ -0,0 +1,114 @@
+package rediscache
+
+import (
+	"time"
+
+	"github.com/go-redis/redis"
+	"github.com/terranodo/tegola/cache"
+	"github.com/terranodo/tegola/util/dict"
+)
+
+const CacheType = "redis"
+
+const (
+	ConfigKeyNetwork  = "network"
+	ConfigKeyAddress  = "address"
+	ConfigKeyPassword = "password"
+	ConfigKeyDB       = "db"
+	ConfigKeyMaxZoom  = "max_zoom"
+)
+
+func init() {
+	cache.Register(CacheType, New)
+}
+
+func New(config map[string]interface{}) (rcache cache.Interface, err error) {
+
+	// default values
+	defaultNetwork := "tcp"
+	defaultAddress := "127.0.0.1:6379"
+	defaultPassword := ""
+	defaultDB := 0
+	defaultMaxZoom := 0
+
+	c := dict.M(config)
+
+	network, err := c.String(ConfigKeyNetwork, &defaultNetwork)
+	if err != nil {
+		return
+	}
+
+	addr, err := c.String(ConfigKeyAddress, &defaultAddress)
+	if err != nil {
+		return
+	}
+
+	password, err := c.String(ConfigKeyPassword, &defaultPassword)
+	if err != nil {
+		return
+	}
+
+	db, err := c.Int(ConfigKeyDB, &defaultDB)
+	if err != nil {
+		return
+	}
+
+	client := redis.NewClient(&redis.Options{
+		Network:     network,
+		Addr:        addr,
+		Password:    password,
+		DB:          db,
+		PoolSize:    2,
+		DialTimeout: 3 * time.Second,
+	})
+
+	pong, err := client.Ping().Result()
+	if err != nil || pong != "PONG" {
+		return
```

Comment:
change to the non-bare version:
`return nil, err`

Should this return an error if pong != "PONG". What does that mean? We are sending back a nil for the client, but also, a nil error here? How, is the client suppose to deal with this?

Reply:
This should be changed. The redis `ping` command should return the string `PONG`, receiving something else shouldn't happen.